src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_pending_resumption_handler.cc

```diff
@@ -0,0 +1,234 @@
+
+RCPendingResumptionHandler::RCPendingResumptionHandler(
+    application_manager::ApplicationManager& application_manager)
+    : ExtensionPendingResumptionHandler(application_manager)
+    , rpc_service_(application_manager.GetRPCService()) {}
+
+void RCPendingResumptionHandler::on_event(
+    const application_manager::event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const auto cid = event.smart_object_correlation_id();
+  LOG4CXX_TRACE(logger_,
+                "Received event with function id: "
+                    << event.id() << " and correlation id: " << cid);
+  sync_primitives::AutoLock lock(pending_resumption_lock_);
+  const auto& data_request = pending_requests_.find(cid);
+  if (data_request == pending_requests_.end()) {
+    LOG4CXX_ERROR(logger_,
+                  "Not waiting for message with correlation id: " << cid);
+    return;
+  }
+
+  auto current_request = data_request->second;
+  pending_requests_.erase(data_request);
+  auto module_uid = GetModuleUid(current_request);
+
+  auto& response = event.smart_object();
+
+  if (RCHelpers::IsResponseSuccessful(response)) {
+    LOG4CXX_DEBUG(logger_,
+                  "Resumption of subscriptions is successful"
+                      << " module type: " << module_uid.first
+                      << " module id: " << module_uid.second);
+    HandleSuccessfulResponse(event, module_uid);
+  } else {
+    LOG4CXX_DEBUG(logger_,
+                  "Resumption of subscriptions is NOT successful"
+                      << " module type: " << module_uid.first
+                      << " module id: " << module_uid.second);
+    ProcessNextFreezedResumption(module_uid);
+  }
+}
+
+void RCPendingResumptionHandler::HandleResumptionSubscriptionRequest(
+    application_manager::AppExtension& extension,
+    resumption::Subscriber& subscriber,
+    application_manager::Application& app) {
+  UNUSED(extension);
+  LOG4CXX_AUTO_TRACE(logger_);
+  sync_primitives::AutoLock lock(pending_resumption_lock_);
+  LOG4CXX_TRACE(logger_, "app id " << app.app_id());
+
+  auto rc_extension = RCHelpers::GetRCExtension(app);
+  auto subscriptions = rc_extension->InteriorVehicleDataSubscriptions();
+
+  std::vector<ModuleUid> already_pending;
+  std::vector<ModuleUid> need_to_subscribe;
+  for (auto subscription : subscriptions) {
+    if (IsPendingForResponse(subscription)) {
+      already_pending.push_back(subscription);
+    } else {
+      need_to_subscribe.push_back(subscription);
+    }
+  }
+
+  for (auto subscription : already_pending) {
+    const auto cid = application_manager_.GetNextHMICorrelationID();
+    const auto subscription_request =
+        CreateSubscriptionRequest(subscription, cid);
+    const auto fid = GetFunctionId(*subscription_request);
+    const auto resumption_request =
+        MakeResumptionRequest(cid, fid, *subscription_request);
+    freezed_resumptions_[subscription].push(resumption_request);
+    subscriber(cid, resumption_request);
```

Comment:
@IGapchuk it should be 
```suggestion
    subscriber(app.app_id(), resumption_request);
```